### PR TITLE
Track timestamp of last successfully uploaded treatment from pump

### PR DESCRIPTION
**Pull request by ps2**
Used to specify how far back in time we need to look for events from the pump

### CODE REVIEW

File: NightscoutUploadKit/NightscoutUploader.swift [hunk 1]
```diff
@@ -31,11 +31,17 @@ public class NightscoutUploader: NSObject {
...
+    var treatmentsQueue: [NightscoutTreatment]
     
     var lastMeterMessageRxTime: NSDate?
     
-    var observingPumpEventsSince: NSDate
+    public private(set) var observingPumpEventsSince: NSDate
+    
+    var lastStoredTreatmentTimestamp: NSDate = NSDate.distantPast() {
+        didSet {
+            NSUserDefaults.standardUserDefaults().lastStoredTreatmentTimestamp = lastStoredTreatmentTimestamp
```
Comment (loudnate) [inline NightscoutUploadKit/NightscoutUploader.swift:42]: This is problematic if the user changes their pump ID, no?

**Comment by loudnate** [comment]
So I also plan to solve this problem in LoopKit's DoseStore, which will have a delegate protocol to trigger uploading records from the SQLite database, and maintains an `uploaded` flag. 

We can revisit when we get that far.

File: NSUserDefaults.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//
+//  NSUserDefaults.swift
+//  RileyLink
```
Comment (loudnate) [inline NSUserDefaults.swift]: This file should be in the NightscoutUploadKit framework

Comment (ps2) [inline NSUserDefaults.swift]: Thanks, good catch.

**Comment by loudnate** [comment]
LGTM

**Pull request merged**